### PR TITLE
Plans redesign: Add new toggle design

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR implements the experiment that redesigns the signup flow plans step. Users in the treatment group will see the new toggle section for monthly/annual plans.

NOTE: 
1. This PR uses the changes in https://github.com/Automattic/wp-calypso/pull/50040 as the base branch and is built on top of it.

**Important links**
1. [Figma design](https://www.figma.com/file/NhFdwnznk6LEuVtvc18qRm/LOHP-pricing-test?node-id=5%3A44)
2. pau2Xa-2yE-p2 for the project thread.

Kapture 2021-02-18 at 18 17 31

#### Testing instructions

* Open http://calypso.localhost:3000/ and assign yourself to treatment of the `signup_plans_step_redesign_v1` experiment in the Abacus tool.
* Begin signup at `/start` and verify that the plans step UI matches the [Figma design](https://www.figma.com/file/NhFdwnznk6LEuVtvc18qRm/LOHP-pricing-test?node-id=5%3A44)
* Confirm that in Calypso's /plans view, the UI is unchanged.
* Also confirm that the launch flow UI is unchanged.
* Verify that control experience is unchanged.
 

Related to #

### CODE REVIEW

File: client/lib/plans/features-list.js [hunk 1]
```diff
@@ -253,6 +259,12 @@ export const FEATURES_LIST = {
 			),
 	},
 
+	[ constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR ]: {
+		getSlug: () => constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR,
+		getTitle: () => i18n.translate( 'Advanced SEO (Search Engine Optimisation) tools' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Advanced SEO (Search Engine Optimization) tools' )` *ES Score: 6*

File: client/lib/plans/features-list.js [hunk 2]
```diff
@@ -542,6 +567,14 @@ export const FEATURES_LIST = {
 			),
 	},
 
+	[ constants.FEATURE_INSTALL_PLUGINS ]: {
+		getSlug: () => constants.FEATURE_INSTALL_PLUGINS,
+		getTitle: () =>
+			i18n.translate(
+				'Access to more than 50,000 WordPress plugins to extend functionality for your site'
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Install plugins and extend functionality for your site with access to more than 50,000 WordPress plugins' )` *ES Score: 8*

File: client/my-sites/plan-features-comparison/header.jsx [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+		if ( isInSignup && isMonthlyPlan && annualPricePerMonth < rawPrice ) {
+			const discountRate = Math.round( ( 100 * ( rawPrice - annualPricePerMonth ) ) / rawPrice );
+			return translate( `Save %(discountRate)s%% by paying annually`, { args: { discountRate } } );
+		}
+
+		if ( isInSignup && ! isMonthlyPlan ) {
+			const annualPriceObj = getCurrencyObject( rawPriceAnnual, currencyCode );
+			const annualPriceText = `${ annualPriceObj.symbol }${ annualPriceObj.integer }`;
+
+			return translate( 'billed as %(price)s annually', {
```
Comment (a8ci18n) [inline client/my-sites/plan-features-comparison/header.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **13** times:
`translate( 'billed as %s annually' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-toggle-design)

**Review by dzver — APPROVED** [review]
LGTM

**Pull request merged**